execution/executor/src/lib.rs

```diff
@@ -472,6 +485,9 @@ impl<V: VMExecutor> ChunkExecutor for Executor<V> {
         // carrying any epoch change LI.
         epoch_change_li: Option<LedgerInfoWithSignatures>,
     ) -> Result<Vec<ContractEvent>> {
+        // Reset cache each time before committing chunk to guarantee up-to-date sync info.
+        self.reset_cache();
```

Comment:
This feels tricky (sounds like we are opening up to race conditions), can you elaborate why it's needed?

Reply:
okay, I see " they may be out-of-sync after the node switch back and forth between sync mode and consensus mode." in the summary. Does it mean the it's guaranteed that the two modes won't ever run at the same time? in that case we should do the reset on mode switch, or (maybe better) construct new executor instance on switch.

Reply:
There is no such a place that says something like `switch(MODE)` so I prefer a simple way here.

Reply:
Right that's the spirit we run things in the TCB! 🚤

Reply:
StateComputer.sync_to is where the mode switch happens

Reply:
@zekun000 but the one in state_sync is not that clear. I found the place but I don't think it is worth add a new api to executor_proxy and call `reset` there.

Reply:
you don't have to reset in executor_proxy if you did it in state computer? probably don't need reset in ChunkExecutor at all
```
fn sync_to() {
  self.executor.reset_cache();
  let result = self.synchronizer.sync_to(target).await;
  self.executor.reset_cache();
  result
}

Reply:
@zekun000 I have to reset the executor in state_computer. There are two executors now... You code snippet only applies to the BlockExecutor.

Reply:
why ChunkExecutor needs any cache? it can basically hide in `execute_and_commit_chunk`?

Reply:
Yes. But @msmouse is not happy with that.

Reply:
What?